libs/application/templates/new-primary-school/src/fields/LanguageSelection/index.tsx

```diff
@@ -0,0 +1,156 @@
+import { FieldBaseProps } from '@island.is/application/types'
+import React, { FC, useEffect, useState } from 'react'
+
+import { Box, Button, Stack, Text } from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import { SelectController } from '@island.is/shared/form-fields'
+import { getAllLanguageCodes } from '@island.is/shared/utils'
+import { newPrimarySchoolMessages } from '../../lib/messages'
+import { getApplicationAnswers } from '../../lib/newPrimarySchoolUtils'
+import { getErrorViaPath } from '@island.is/application/core'
+
+const languagesIds = {
+  language1: 'languages.language1',
+  language2: 'languages.language2',
+  language3: 'languages.language3',
+  language4: 'languages.language4',
+  childLanguage: 'languages.childLanguage',
+}
+
+const LanguageSelection: FC<React.PropsWithChildren<FieldBaseProps>> = ({
+  application,
+  errors,
+}) => {
+  const { formatMessage } = useLocale()
+
+  const [visibleLanguagesCount, setVisibleLanguagesCount] = useState(1)
+  const [selectedLanguages, setSelectedLanguages] = useState<string[]>([
+    '',
+    '',
+    '',
+    '',
+  ])
+
+  const allLanguages = getAllLanguageCodes()
+  const allLanguageOptions = allLanguages.map((language) => ({
+    label: language.name,
+    value: language.code,
+  }))
+
+  const languageIdsArray = [
+    languagesIds.language1,
+    languagesIds.language2,
+    languagesIds.language3,
+    languagesIds.language4,
+  ]
+
+  const addLanguage = () => {
+    setVisibleLanguagesCount((prev) => Math.min(prev + 1, 4))
+  }
+
+  /**
+   * Hide child language if there is no language selected
+   * @returns
+   */
+  const hideChildLanguage = () => {
+    return selectedLanguages.filter((language) => language !== '').length <= 1
+  }
+
+  useEffect(() => {
+    const { language1, language2, language3, language4 } =
+      getApplicationAnswers(application.answers)
+    const selected: string[] = []
+    let visibleCount = 1
+    if (language1) {
+      selected.push(language1)
+      visibleCount = 1
+    }
+    if (language2) {
+      selected.push(language2)
+      visibleCount = 2
+    }
+    if (language3) {
+      selected.push(language3)
+      visibleCount = 3
+    }
+    if (language4) {
+      selected.push(language4)
+      visibleCount = 4
+    }
```

Comment:
This can be simplified
```suggestion
    const selected = [language1, language2, language3, language4].filter(Boolean)
    const visibleCount = Math.max(1, selected.length)
```

Reply:
Fixed